test/automated/picolibrary/adafruit/pid781/driver/main.cc

```diff
@@ -0,0 +1,1134 @@
+    /**
+     * \brief The pin whose initial pin state is to be set and saved.
+     */
+    Pin pin;
+
+    /**
+     * \brief The desired initial pin state.
+     */
+    Initial_Pin_State initial_pin_state;
+
+    /**
+     * \brief The unsigned byte written to the stream (pin).
+     */
+    std::uint8_t value_pin;
+
+    /**
+     * \brief The unsigned byte written to the stream (initial_pin_state).
+     */
+    std::uint8_t value_initial_pin_state;
+};
+
+auto operator<<( std::ostream & stream, setAndSaveOutputPinInitialState_Test_Case const & test_case )
+    -> std::ostream &
+{
+    // clang-format off
+
+    return stream << "{ "
+                  << ".pin = " << test_case.pin
+                  << ", "
+                  << ".initial_pin_state = " << test_case.initial_pin_state
+                  << ", "
+                  << ".value_pin = " << static_cast<std::uint_fast16_t>( test_case.value_pin )
+                  << ", "
+                  << ".value_initial_pin_state = " << static_cast<std::uint_fast16_t>( test_case.value_initial_pin_state )
+                  << " }";
+
+    // clang-format on
+}
+
+/**
+ * \brief picolibrary::Adafruit::PID781::Driver::set_and_save_output_pin_initial_state()
+ *        test fixture.
+ */
+class setAndSaveOutputPinInitialState :
+    public TestWithParam<setAndSaveOutputPinInitialState_Test_Case> {
+};
+
+/**
+ * \brief Verify
+ *        picolibrary::Adafruit::PID781::Driver::set_and_save_output_pin_initial_state()
+ *        works properly.
+ */
+TEST_P( setAndSaveOutputPinInitialState, worksProperly )
+{
+    auto const test_case = GetParam();
+
+    auto pid781 = Driver{};
+
+    pid781.set_and_save_output_pin_initial_state( test_case.pin, test_case.initial_pin_state );
+
+    ASSERT_EQ(
+        pid781.vector(),
+        ( std::vector<std::uint8_t>{
+            0xFE,
+            0xC3,
+            test_case.value_pin,
+            test_case.value_initial_pin_state,
+        } ) );
+}
+
+/**
+ * \brief picolibrary::Adafruit::PID781::Driver::set_and_save_output_pin_initial_state()
+ *        test cases.
+ */
+setAndSaveOutputPinInitialState_Test_Case const setAndSaveOutputPinInitialState_TEST_CASES[]{
+    // clang-format off
+
+    { Pin::PB0, Initial_Pin_State::LOW,  1, 0 },
+    { Pin::PB0, Initial_Pin_State::HIGH, 1, 1 },
+
+    { Pin::PC2, Initial_Pin_State::LOW,  2, 0 },
+    { Pin::PC2, Initial_Pin_State::HIGH, 2, 1 },
+
+    { Pin::PC4, Initial_Pin_State::LOW,  3, 0 },
+    { Pin::PC4, Initial_Pin_State::HIGH, 3, 1 },
+
+    { Pin::PC7, Initial_Pin_State::LOW,  4, 0 },
+    { Pin::PC7, Initial_Pin_State::HIGH, 4, 1 },
+
+    // clang-format on
+};
+
+INSTANTIATE_TEST_SUITE_P( testCases, setAndSaveOutputPinInitialState, ValuesIn( setAndSaveOutputPinInitialState_TEST_CASES ) );
+
+/**
+ * \brief picolibrary::Adafruit::PID781::Driver::transition_output_pin_to_high() test
+ *        case.
+ */
+struct transitionOutputPinToHigh_Test_Case {
```

Comment:
`::picolibrary::Adafruit::PID781::transition_output_pin_to_high()` and `::picolibrary::Adafruit::PID781::transition_output_pin_to_low()` can use the same test case/cases.